### DISCUSSION TITLE
Codegen Subscription URQL

### DISCUSSION THREAD

**Discussion by enri90** [hayes/pothos#893] [General]
how do i do urql codegen with subscriptions? 
I use this plugin
https://pothos-graphql.dev/docs/plugins/smart-subscriptions

i get this message

```typescript
> graphql-codegen --config codegen.ts

(0 , _pluginSmartSubscriptions.subscribeOptionsFromIterator) is not a function

``` 

```typescript
'../test/modules/production/graphql/': {
      preset: 'near-operation-file',
      presetConfig: {
        baseTypesPath: '../../../graphql/types.ts',
        folder: '../../../test/modules/production/graphql',
        extension: '.ts',
      },
      plugins: [
        {
          add: {
            content: '/* eslint-disable */',
          },
        },
        'typescript-operations',
        'typescript-urql',
      ],
      config: {
        documentMode: 'documentNode',
        immutableTypes: false,
        withComponent: false,
        withHooks: true,
        enumsAsConst: true,
        subscriptionReturnType: 'Promise<{ data?: TData; error?: Error }>',
        subscriptionResultTypeName: 'TData',
        namingConvention: {
          enumValues: 'upper-case#upperCase',
        },
      },
    },
```

**Comment by hayes** [comment]
This looks like an issue with how graphql-codegen is bundling your schema.  There shouldn't be anything specific to subscriptions that causes issues.

It doesn't look like you've shared how the codegen config imports the schema, or how the builder is set up.  This looks to me like it could be some sort of an esm/cjs interop issue.   

`_pluginSmartSubscriptions.subscribeOptionsFromIterator` looks to me like `_pluginSmartSubscriptions` is supposed to be the plugin module, and it's reading `subscribeOptionsFromIterator` from that module.  The pothos plugin does export that and it is a function.  

Try ensuring that you enable https://www.typescriptlang.org/tsconfig#esModuleInterop in your tsconfig (not sure if this will fix your issue, but that's my best guess)
